app/lib/search/text_utils.dart

```diff
@@ -110,8 +102,39 @@ Map<String, double> tokenize(String originalText) {
   return tokens;
 }
 
+// Scan for CamelCase phrases and extract Camel and Case separately.
+Iterable<String> extractCamelCase(String word) sync* {
+  final changeIndex = <int>[0];
+  bool prevLower = _isLower(word[0]);
+  for (int i = 1; i < word.length; i++) {
+    final bool lower = _isLower(word[i]);
+    if (!lower && prevLower) {
+      changeIndex.add(i);
+    }
+    prevLower = lower;
+  }
+  changeIndex.add(word.length);
+  for (int i = 1; i < changeIndex.length; i++) {
+    final token = word.substring(changeIndex[i - 1], changeIndex[i]);
+    yield token;
+  }
+}
+
 bool _isLower(String c) => c.toLowerCase() == c;
```

Comment:
Not sure if this is a bottle-neck, but considering that dart-identifiers are ascii-only this could done with:
```dart
  const lowerA = 97;
  const lowerZ = 122;
  bool _isLower(int char) => lowerA <= char && char <= lowerZ;
```